mypy/messages.py

```diff
@@ -1077,8 +1078,21 @@ def unimported_type_becomes_any(self, prefix: str, typ: Type, ctx: Context) -> N
         self.fail("{} becomes {} due to an unfollowed import".format(prefix, self.format(typ)),
                   ctx)
 
-    def need_annotation_for_var(self, node: SymbolNode, context: Context) -> None:
-        self.fail("Need type annotation for '{}'".format(unmangle(node.name())), context)
+    def need_annotation_for_var(self, node: SymbolNode, context: Context,
+                                python_version: Tuple[int, int]) -> None:
+        hint = ''  # Only gives hint if it's a variable declaration and the partial type is a builtin type
+        if (isinstance(node, Var) and isinstance(node.type, PartialType) and
+                node.type.type and node.type.type.fullname() in reverse_builtin_aliases):
+            alias = reverse_builtin_aliases[node.type.type.fullname()]
+            alias = alias.split('.')[-1]
+            type_dec = '<type>'
+            if alias == 'Dict':
+                type_dec = '{}, {}'.format(type_dec, type_dec)
+            if python_version < PYTHON3_VERSION:
```

Comment:
I'd prefer `... < (3, 6)` here since `PYTHON3_VERSION` might get changed to `(3, 7)` in the future and the would then become incorrect.